configs/app/features/marketplace.ts

```diff
@@ -22,31 +25,38 @@ const config: Feature<(
   categoriesUrl: string | undefined;
   suggestIdeasFormUrl: string | undefined;
   securityReportsUrl: string | undefined;
-}
-> = (() => {
+  featuredApp: string | undefined;
+  banner: { contentUrl: string; linkUrl: string } | undefined;
+}> = (() => {
   if (enabled === 'true' && chain.rpcUrl && submitFormUrl) {
+    const props = {
+      submitFormUrl,
+      categoriesUrl,
+      suggestIdeasFormUrl,
+      securityReportsUrl,
+      featuredApp,
+      banner: bannerContentUrl && bannerLinkUrl ? {
+        contentUrl: bannerContentUrl,
+        linkUrl: bannerLinkUrl,
+      } : undefined,
```

Comment:
Yes, they are mutually exclusive. If you pass all of them you will see a featured app: https://github.com/blockscout/frontend/pull/1785/files#diff-8a909d3cb91cceb61e336bae27add42d37e5c2126499f2ec5c58adc13278cd78R28

Reply:
So maybe it will be better to make a compound type, like `{ featuredApp: string; } |  { banner: ... }`. Or it is too complex and not necessary? What do you think?

Reply:
That's what I originally started doing, but it ended up with more code and looked less clear in my opinion.
And you can't just check `feature.banner` when using it, you have to check `'banner' in feature`, which is also less clear in my opinion.

Reply:
Fair enough, ok. Let's leave it as it is then.